opensrp-reveal/src/main/java/org/smartregister/reveal/sync/RevealClientProcessor.java

```diff
@@ -103,14 +104,17 @@ public void processClient(List<EventClient> eventClients, boolean localEvents) {
                     }
                 }
             }
-        }
 
+            if (!hasSyncedEventsInTarget && operationalAreaLocationId != null &&
```

Comment:
This must be executed inside the events for loop. So that each event is evaluated

Reply:
Could you look at this again and rephrase what the issue is?

Isn't the code still executed in the for loop?

Reply:
This is not inside the for loop, for loop ends at line 106

Reply:
Ok.